### DISCUSSION TITLE
What ports are needed to be open?

### DISCUSSION THREAD

**Discussion by SemoTech** [w2c/letsencrypt-esxi#22] [Q&A]
Can this work with just port 443 (HTTPS) open through the firewall and pointed to the ESXi host, or does it need both 80 (HTTP) and 443 (HTTPS) open?

**Comment by Churro** [comment]
It needs port 80 as well. Let's Encrypt relies on this port to do a validation, see https://letsencrypt.org/docs/allow-port-80/

**Reply by SemoTech** [reply]
Gotcha, thanks @Churro, but I thought ESXi automatically redirects 80 to 443, does this plugin change that?

**Reply by Churro** [reply]
No, this plugin doesn't interfere with port mappings. It only adds an HTTP route to handle Let's Encrypt's `.well-known/acme-challenge` requests. Let's Encrypt requests port 80 specifically, and therefore this port needs to be reachable.

**Reply by jacobswin** [reply]
Is that possible to use any other port as the 80 and 443 is blocked by ISP?

**Reply by Churro** [reply]
No, not possible. This isn't a limitation of this plugin but enforced by Let's Encrypt itself.

**Reply by jacobswin** [reply]
Uh, I understand. If the script can run by DNS challenge, that will be great!

**Reply by Churro** [reply]
I'm with you but that's quite tricky, unfortunately -> see https://github.com/w2c/letsencrypt-esxi/issues/2
